arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/NonEmptySet.kt

```diff
@@ -0,0 +1,34 @@
+package arrow.core
+
+public class NonEmptySet<out T>(
+  private val first: T,
+  private val rest: Set<T>,
+) : AbstractSet<T>() {
+
+  public val elements: Set<T> by lazy { setOf(first) + rest }
+
+  public operator fun plus(other: NonEmptySet<@UnsafeVariance T>): NonEmptySet<T> =
+    NonEmptySet(this.first, rest + other.elements)
```

Comment:
Is this overload necessary, or is the `plus(Set)` function sufficient for both cases?